app/Notifications/WelcomeEmailNotification.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace App\Notifications;
+
+use App\Channels\SailthruChannel;
+use App\User;
+use Illuminate\Bus\Queueable;
+use Illuminate\Notifications\Notification;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Notifications\Messages\MailMessage;
+
+class WelcomeEmailNotification extends Notification
+{
+    use Queueable;
+
+    /**
+     * @var string
+     */
+    public $sailThruTemplate = '';
+
+    /**
+     * @var array
+     */
+    public $sailThruParameters = [];
+
+    /**
+     * Create a new notification instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+    }
+
+    /**
+     * Get the notification's delivery channels.
+     *
+     * @param  mixed  $notifiable
+     * @return array
+     */
+    public function via($notifiable)
+    {
+        return [
+            'mail',
+            SailthruChannel::class
+        ];
```

Comment:
I would make it explicit here since each notification is defined in a known context. 

So this could be something like 

```
if (domain_config('sailthru.enabled')) {
   return SailthruChannel::class;
} else {
   return ['mail'];
} 


```

If this was an admin notification like `Admin\ReportAbuseNotification`, we'd just hardcode it to go  by mail,
```
return ['mail']

```
 or use one single config
```
return domain_config('mail.admin_driver')
```
or via multiple channels like 
```
return [domain_config('mail.admin_driver'), 'slack']

```

By being explicit within the notification, we can drop the need for complicated configs and magic.